### ISSUE TITLE
[FR]: Add a built-in updater for AppImages

### ISSUE THREAD

**Issue by Mitecon** [martinrotter/rssguard#975]
### Brief description of the feature request

I find I use RSS Guard Appimages more than anything else because they're always more up to date than the versions in my repo.

I also use [QOwnNotes](https://github.com/pbek/QOwnNotes/) which has a built-in AppImage updating function that works in conjunction with [AppImageLauncher](https://github.com/TheAssassin/AppImageLauncher). Whenever a new version is detected, I can click on the notification in the status bar and a nice window pops up with various information:

![qownnotes-updater-changelog](https://github.com/martinrotter/rssguard/assets/125831943/b7a10a2c-de40-4fc0-aaee-b1d3d5ea40cf)

There are various buttons to do different things. I normally just click 'Update':

![qownnotes-updater-update-in-progress](https://github.com/martinrotter/rssguard/assets/125831943/8cf1cedb-1005-469b-aaf8-011040b52850)

Then, when the update is complete, I get this dialogue:

![qownnotes-updater-update-complete](https://github.com/martinrotter/rssguard/assets/125831943/371da35e-0fda-4509-b02a-777b8f9d61f5)

What's nice is that QOwnNotes simply replaces the old AppImage.

At the moment, updating RSS Guard is all manual but it also leads to this situation when searching for it in the launcher:

![rss-guard-appimages-in-launcher](https://github.com/martinrotter/rssguard/assets/125831943/179f2c53-30b8-44a0-90a8-71aa6808a7ba)

This means you have to go hunting around to see which one you can launch and which one you can uninstall. Obviously, people shouldn't be downloading a new version with bugfixes, only to accidentally continue running the old version and complain that the bug isn't fixed.

Plus, you could have options like 'Install latest stable version' and 'Install latest devbuild'.

There's information here:

https://github.com/AppImageCommunity/AppImageUpdate#update-information-overview

While the QOwnNotes Issues discussing it are here:

https://github.com/pbek/QOwnNotes/issues/1991
https://github.com/pbek/QOwnNotes/issues/2176

This would be a 'nice to have'.

**Comment by martinrotter** [comment]
Yes, this could indeed be added. Note that there is already "update" dialog in RSS Guard which supports something similar for Windows. Appimage support could be indeed added.

**Comment by martinrotter** [comment]
Not planning adding this myself, PRs are welcomed.
